### ISSUE TITLE
[TRAINER] Create Base Trainer Class

### ISSUE THREAD

**Issue by art-test-stack** [art-test-stack/vessel_position_forecast#7]
init inputs: 
- model: Can be: XGBoost, torch.nn.Module (AutoRegressive Transformer, LSTM, etc.)
- loss
- metric (for now the loss will be sufficient) -> have to be custom metric in the future
- optimizer (if neural network)
- device
- maybe some others
